test/porter/inspect_test.rb

```diff
@@ -0,0 +1,58 @@
+require 'test_helper'
+require 'aws-sdk-states'
+require 'json'
+
+step_functions = Aws::States::Client.new(
+  region: 'us-east-1'
+)
+
+describe :porter do
+  describe :inspect do
+    it 'returns the basic execution output' do
+      req = step_functions.start_execution({
+        state_machine_arn: CONFIG.PORTER_STATE_MACHINE_ARN,
+        input: {
+          Job: {
+            Id: 'porter-test-inspect',
+            Source: {
+              Mode: 'HTTP',
+              URL: 'https://media.prx.org/emailimages/ieeespectrum.png'
+            },
+            Tasks: [
+              {
+                Type: "Inspect"
+              }
+            ]
+          }
+        }.to_json
+      })
+
+      max_retries = 5
+      retries = 0
+
+      begin
+        desc = step_functions.describe_execution({
+          execution_arn: req.execution_arn,
+        })
+
+        raise RuntimeError if desc.status == 'RUNNING'
+      rescue RuntimeError => e
+        if retries <= max_retries
+          retries += 1
+          sleep 2 ** retries
```

Comment:
Hrmm... probably don't want an exponential backoff here?  Seeing these taking quite some time locally (16, 32 or even 64 seconds each).  When the actual porter execution is < 2 seconds.

I landed on more like `max_retries = 10` and just a half second `sleep 0.5`.

Reply:
@cavis Given that 10 half-second retries would be a total of 5 seconds, and the backoff gives a total of 6 seconds (assuming it takes more than 2 seconds), the backoff seems safer, since it also handles these cases where it takes much longer than 5 seconds.

Reply:
Maybe just a bunch more retries then?  Like `max_retries = 60` and just `sleep 1` every time, if 60 seconds is a valid amount of time for this to take?

There's no real cost to hitting the AWS API more frequently, so I don't really think the backoff is useful.  And the benefit is the test finishes much faster in some cases.  If the actual execution takes 33 seconds (and these often seem to land just over 30 seconds), this code currently would waste 31 seconds because it didn't check frequently enough.